src/game/SpellEffects.cpp

```diff
@@ -3835,7 +3835,7 @@ bool Spell::DoSummonPet(SpellEffectIndex eff_idx)
     {
         // Load pet from db; if any to load
         if (m_caster->getClass() != CLASS_PRIEST
```

Comment:
This may be something that changes with the rework of EffectSummonType, because priests and mage's don't summon their pets via SummonPet or SummonGuardian, they summon them via EffectSummon . (at least in vanilla / tbc) - if they now go through SummonPet (which is fine if they are) then you need to surpress them getting saved to the db, for consistency you know, this....